src/Adapter/Product/Stock/Validate/StockAvailableValidator.php

```diff
@@ -71,4 +73,11 @@ private function validateStockAvailableProperty(StockAvailable $stockAvailable,
             $errorCode
         );
     }
+
+    private function validateMaxQuantity(int $quantity)
```

Comment:
I believe that now that the ObjectModel validation has been improved this custom validation is no longer useful, since this check is already performed via `$this->validateStockAvailableProperty($stockAvailable, 'quantity', ProductStockConstraintException::INVALID_QUANTITY);`

You can make sure of it simply by commenting this code and running the behat test you added for this use case, they should still be green meaning this code can be removed.